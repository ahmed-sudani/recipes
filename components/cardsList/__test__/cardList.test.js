import { render, screen, within } from '@testing-library/react'
import CardsList from '..'

const cards = [
  {
    id: '1',
    name: 'chechen',
    time: '50:00',
    img: '/cster.png',
    country: 'italy',
  },

  {
    id: '2',
    name: 'chechen',
    time: '50:00',
    img: '/cster.png',
    country: 'italy',
  },
]

describe('Cards list', () => {
  it('should render CardList component', () => {
    render(<CardsList cards={[]} />)
    const cardListElement = screen.getByTestId('container')
    expect(cardListElement).toBeDefined()
  })
  it('should have 2 cards in the list', () => {
    render(<CardsList cards={cards} />)
    const cardListElement = screen.getByTestId('container')
    const cardsList = within(cardListElement).getAllByTestId('card')
    expect(cards.length).toEqual(2)
  })
})
