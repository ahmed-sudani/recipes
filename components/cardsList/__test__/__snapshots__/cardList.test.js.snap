// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards list should render CardList component 1`] = `
<div
  class="container"
  data-testid="container"
/>
`;
